src/Faker/UniqueGenerator.php

```diff
@@ -3,13 +3,27 @@
 namespace Faker;
 
 /**
- * Proxy for other generators, to return only unique values. Works with
- * Faker\Generator\Base->unique()
+ * Proxy for other generators that returns only unique values.
+ *
+ * Instantiated through @see Generator::unique().
+ *
+ * @mixin Generator
  */
 class UniqueGenerator
 {
+    /**
+     * @var Generator
+     */
     protected $generator;
+
+    /**
+     * @var int
+     */
     protected $maxRetries;
+
+    /**
+     * @var array<string, array<string, null>>
```

Comment:
This is not valid i assume? In fact there could be dates / int / string returned and so on if im correct right?

Reply:
I left an explanatory comment for you.

Reply:
But the array can hold more then just an array with strings only if im correct

Reply:
```
$this->uniques[$name] = [];
$this->uniques[$name][serialize($res)] = null;
```

These are the only two writes to the array.